### PR TITLE
Gather datastore metrics using go-metrics-interface

### PR DESCRIPTION
**Pull request by polezaivsani**

### CODE REVIEW

File: measure.go [hunk 1]
```diff
@@ -1,54 +1,55 @@
...
-
-		queryNum:     metrics.Counter(prefix + ".Query.num"),
-		queryErr:     metrics.Counter(prefix + ".Query.err"),
-		queryLatency: metrics.NewHistogram(prefix+".Query.latency", 0, maxLatency, 3),
+		putNum: metrics.New(prefix+".put.num", "Total number of Datastore.Put calls").Counter(),
+		putErr: metrics.New(prefix+".put.err", "Number of errored Datastore.Put calls").Counter(),
+		putLatency: metrics.New(prefix+".put.latency",
+			"Latency distribution of Datastore.Put calls").Histogram(datastoreLatencyBuckets),
+		putSize: metrics.New(prefix+".put.size",
+			"Stored []byte length distribution").Histogram(datastoreLatencyBuckets),
```
Comment (Kubuxu) [inline measure.go]: You are using the same buckets for latency and size, I would suggest buckets like `1<<6, 1<<12, 1<<18, 1<<24` or similar for the size.

**Comment by polezaivsani** [comment]
Stupid me. Fixed.

File: measure.go [hunk 2]
```diff
@@ -58,87 +62,85 @@ type measure struct {
...
 	queryErr     metrics.Counter
-	queryLatency *metrics.Histogram
+	queryLatency metrics.Histogram
 }
 
-var _ datastore.Datastore = (*measure)(nil)
-
-func recordLatency(h *metrics.Histogram, start time.Time) {
+func recordLatency(h metrics.Histogram, start time.Time) {
 	elapsed := time.Now().Sub(start) / time.Microsecond
```
Comment (Kubuxu) [inline measure.go]: Can you replace it with `time.Since()`.

File: measure.go [hunk 3]
```diff
@@ -204,42 +206,24 @@ func (mt *measuredBatch) Commit() error {
 	return nil
 }
 
-func logBatchCommit(b datastore.Batch, n int, num, errs metrics.Counter, lat *metrics.Histogram) error {
+func logBatchCommit(b datastore.Batch, n int, num, errs metrics.Counter, lat metrics.Histogram) error {
 	if n > 0 {
 		before := time.Now()
 		err := b.Commit()
 		took := int(time.Now().Sub(before)/time.Microsecond) / n
```
Comment (Kubuxu) [inline measure.go]: Same in there `time.Since()`.

**Comment by Kubuxu** [comment]
We use Prometheus's convention when it comes to metric names. This means that metrics should end with unit they are expressed in. For example `metric.name_seconds` or if it is unit less `metric.name.errors_total`.

**Comment by Kubuxu** [comment]
LGTM

//cc @whyrusleeping for merge

**Comment by whyrusleeping** [comment]
@polezaivsani @Kubuxu This LGTM, but just for completeness sake, could one of you add a test for this? (I'm also fine with filing an issue to add tests and moving on)

**Comment by Kubuxu** [comment]
I am for the latter, will create issue.

**Comment by whyrusleeping** [comment]
@Kubuxu cool, then lets get this shipped

**Pull request merged**